pkg/uevent/sync.go

```diff
@@ -0,0 +1,237 @@
+func (l *listener) sync(ctx context.Context) error {
+	dir, err := os.Open("/run/udev/data")
+	if err != nil {
+		return err
+	}
+	defer dir.Close()
+
+	names, err := dir.Readdirnames(-1)
+	if err != nil {
+		return err
+	}
+
+	var devices []*sys.Device
+	for _, name := range names {
+		if !strings.HasPrefix(name, "b") {
+			continue
+		}
+		major, minor, err := utils.GetMajorMinorFromStr(strings.TrimPrefix(name, "b"))
+		if err != nil {
+			klog.V(5).Infof("error while parsing maj:min for file: %s: %v", name, err)
+			continue
+		}
+		devName, err := sys.GetDeviceName(major, minor)
+		if err != nil {
+			klog.V(5).Infof("error while getting device name for maj:min (%v:%v): %v", major, minor, err)
+			continue
+		}
+		if sys.IsLoopBackDevice("/dev/" + devName) {
+			klog.V(5).InfoS("loopback device is ignored while syncing", "DEVNAME", devName)
+			continue
+		}
+		data, err := sys.ReadRunUdevDataByMajorMinor(int(major), int(minor))
+		if err != nil {
+			klog.V(5).Infof("error while reading udevdata for device %s: %v", devName, err)
+			continue
+		}
+		runUdevData, err := sys.MapToUdevData(data)
+		if err != nil {
+			klog.V(5).Infof("error while mapping udevdata for device %s: %v", devName, err)
+			continue
+		}
+		device := &sys.Device{
+			Name:         devName,
+			Major:        int(major),
+			Minor:        int(minor),
+			Virtual:      strings.Contains(devName, "virtual"),
+			Partition:    runUdevData.Partition,
+			WWID:         runUdevData.WWID,
+			Model:        runUdevData.Model,
+			UeventSerial: runUdevData.UeventSerial,
+			Vendor:       runUdevData.Vendor,
+			DMName:       runUdevData.DMName,
+			DMUUID:       runUdevData.DMUUID,
+			MDUUID:       runUdevData.MDUUID,
+			PTUUID:       runUdevData.PTUUID,
+			PTType:       runUdevData.PTType,
+			PartUUID:     runUdevData.PartUUID,
+			UeventFSUUID: runUdevData.UeventFSUUID,
+			FSType:       runUdevData.FSType,
+			PCIPath:      runUdevData.PCIPath,
+			SerialLong:   runUdevData.UeventSerialLong,
+		}
+		// Probe from /sys/
+		if err := device.ProbeSysInfo(); err != nil {
+			klog.V(5).Infof("error while probing sys info for device %s: %v", devName, err)
+			continue
+		}
+		// Probe from /proc/1/mountinfo
+		if err := device.ProbeMountInfo(); err != nil {
+			klog.V(5).Infof("error while probing dev info for device %s: %v", devName, err)
+			continue
+		}
+		// Opens the device `/dev/` to probe
+		if err := device.ProbeDevInfo(); err != nil {
+			klog.V(5).Infof("error while validating device %s: %v", devName, err)
+			continue
+		}
+		devices = append(devices, device)
+	}
+
+	return l.syncDevices(ctx, devices)
+}
+
+// Tries to match the directcsidrives by the discovered and probed devices from host
+//
+// follows group based match, managed (ready|inuse) and non-managed (available|unavailable|released),
+// this isolates them and do-not allow matching collision between two groups
+//
+// Managed directcsidrives (InUse and Ready) :-
```

Comment:
IMO this logic of matching only fs attribute  for maned drive should also be a part of architecture.mc docs

Reply:
sure, will add this in the arch docs @sinhaashish